src/nodes/primitive/range.ts

```diff
@@ -146,20 +206,34 @@ export const rangeNode = defineNodeKind<RangeNode>(
         }
     },
     (base) => {
-        const leftDescription = `${base.rule[0].comparator}${base.rule[0].limit}`
-        const description = base.rule[1]
-            ? `the range bounded by ${leftDescription} and ${base.rule[1].comparator}${base.rule[1].limit}`
-            : leftDescription
+        const description = getDescription(base.rule[0], base.rule[1])
+        const min = isKeyOf(base.rule[0].comparator, minComparators)
+            ? (base.rule[0] as Bound<MinComparator>)
+            : undefined
+        const max =
+            base.rule[1] ??
+            (isKeyOf(base.rule[0].comparator, maxComparators)
+                ? (base.rule[0] as Bound<MaxComparator>)
+                : undefined)
         return {
             description,
-            min: isKeyOf(base.rule[0].comparator, minComparators)
-                ? (base.rule[0] as Bound<MinComparator>)
-                : undefined,
-            max:
-                base.rule[1] ??
-                (isKeyOf(base.rule[0].comparator, maxComparators)
-                    ? (base.rule[0] as Bound<MaxComparator>)
-                    : undefined)
+            min,
+            max,
+            numericMin:
+                min && isDate(min.limit)
```

Comment:
You can get rid of all these conditionals and just use `valueOf` in all cases since it's also attached to `number`